discovery/grpc_srv_resolver.go

```diff
@@ -0,0 +1,268 @@
+		if time.Since(lastSuccess) < time.Second*30 {
+			goto wait
+		}
+
+		if err := d.lookupSRV(); err != nil {
+			d.cc.ReportError(err)
+			next := backOff.NextIteration()
+			DefaultLogger.WithError(err).WithField("retry-after", next).
+				Error("dns lookup failed; retrying...")
+
+			timer := time.NewTimer(next)
+			select {
+			case <-d.ctx.Done():
+				timer.Stop()
+				return
+			case <-timer.C:
+			}
+			continue
+		}
+		lastSuccess = time.Now()
+	wait:
+		backOff.Reset()
+
+		select {
+		case <-d.ctx.Done():
+			ticker.Stop()
+			return
+		case <-ticker.C:
+		case <-d.rn:
+		}
+	}
+}
+
+func (d *srvResolver) lookupSRV() error {
+	var result []resolver.Address
+
+	// TODO(thrawn01): At some point in the future we might parse the Target and determine
+	//  if the Target name is in the RFC 2782 form of `_<service>._tcp[.service][.<datacenter>].<domain>`
+	//  then fill out the service and proto fields in LookupSRV()
+
+	_, srvs, err := net.DefaultResolver.LookupSRV(d.ctx, "", "", d.host)
+	if err != nil {
+		return fmt.Errorf("SRV record lookup err: %w", err)
+	}
+	for _, s := range srvs {
+		resolved, err := net.DefaultResolver.LookupHost(d.ctx, s.Target)
+		if err != nil {
+			if herr := handleDNSError(err, "A"); herr == nil {
+				DefaultLogger.WithError(err).WithField("target", s.Target).
+					Error("error resolving 'A' records for SRV entry")
+				continue
+			}
+			return err
+		}
+
+		var addresses []resolver.Address
+		for _, a := range resolved {
+			ip, ok := formatIP(a)
+			if !ok {
+				DefaultLogger.WithField("ip", ip).
+					Error("error parsing 'A' record for SRV entries; is not a valid ip address")
+				continue
+			}
+			addresses = append(addresses, resolver.Address{Addr: ip + ":" + strconv.Itoa(int(s.Port)), ServerName: s.Target})
+		}
+
+		// If our current state is empty, then immediately update state before looking up the remaining SRV records.
+		// Looking up a lot of hosts could take a lot of time, and we want to connect as quickly as possible.
+		// During testing, a DNS lookup on all service nodes for `ratelimits` took 5+ seconds, which caused the
+		// GRPC calls to timeout.
+		if len(d.state.Addresses) == 0 {
+			if err := d.cc.UpdateState(resolver.State{Addresses: addresses}); err != nil {
+				DefaultLogger.WithError(err).Error("UpdateState() call returned an error")
+			}
+			d.state.Addresses = addresses
+		}
+		result = append(result, addresses...)
+	}
+
+	if len(result) == 0 {
+		return fmt.Errorf("SRV record for '%s' contained no valid domain names", d.host)
+	}
+
+	d.state.Addresses = result
+	if GRPCSrvLogAddresses {
+		var addresses []string
+		for _, a := range result {
+			addresses = append(addresses, a.Addr)
+		}
+		DefaultLogger.WithField("addresses", addresses).Info("dns-srv: address list updated")
+	}
+	return d.cc.UpdateState(d.state)
+}
+
+func handleDNSError(err error, lookupType string) error {
+	if dnsErr, ok := err.(*net.DNSError); ok && !dnsErr.IsTimeout && !dnsErr.IsTemporary {
```

Comment:
Please, use `errors.As()`

https://go.dev/blog/go1.13-errors

Reply:
This was code from the original GRPC implementation. I will removed this code, as it is no longer necessary.